Rahul/StatePatternCPP/src/GumballMachine.cpp

```diff
@@ -0,0 +1,65 @@
+#include "GumballMachine.hpp"
+#include "NoQuarterState.hpp"
+#include "HasQuarterState.hpp"
+#include "SoldState.hpp"
+#include "SoldOutState.hpp"
+#include "WinnerState.hpp"
+
+GumballMachine::GumballMachine(int balls){
+    this->gumball = balls;
+    this->noQuarterState = AbstractStatePtr((AbstractState *) new NoQuarterState(GumballMachinePtr(this)) );
```

Comment:
GumballMachinePtr(this) --- this is dangerous...read about std::enable_shared_from_this